core/net/ip/ip64-addr.c

```diff
@@ -36,6 +36,61 @@
 
 #define printf(...)
 /*---------------------------------------------------------------------------*/
+static int
+is_private_ipv4addr(const uip_ip4addr_t *ipv4addr)
+{
+  uip_ipaddr_t ipaddr, mask;
+
+  /* 10.0.0.0/8 */
+  uip_ipaddr(&ipaddr, 10, 0, 0, 0);
+  uip_ipaddr(&mask, 255, 0, 0, 0);
+  if(uip_ipaddr_maskcmp(ipv4addr, &ipaddr, &mask)) {
+    return 1;
+  }
+
+  /* 172.16.0.0/12 */
+  uip_ipaddr(&ipaddr, 172, 16, 0, 0);
+  uip_ipaddr(&mask, 255, 240, 0, 0);
+  if(uip_ipaddr_maskcmp(ipv4addr, &ipaddr, &mask)) {
+    return 1;
+  }
+
+  /* 198.18.0.0/15 */
+  uip_ipaddr(&ipaddr, 198, 18, 0, 0);
+  uip_ipaddr(&mask, 255, 254, 0, 0);
+  if(uip_ipaddr_maskcmp(ipv4addr, &ipaddr, &mask)) {
+    return 1;
+  }
+
+  /* 192.168.0.0/16 */
+  uip_ipaddr(&ipaddr, 192, 168, 0, 0);
+  uip_ipaddr(&mask, 255, 255, 0, 0);
+  if(uip_ipaddr_maskcmp(ipv4addr, &ipaddr, &mask)) {
+    return 1;
+  }
+
+  /* 192.0.2.0/24 */
+  uip_ipaddr(&ipaddr, 192, 0, 2, 0);
+  uip_ipaddr(&mask, 255, 255, 255, 0);
+  if(uip_ipaddr_maskcmp(ipv4addr, &ipaddr, &mask)) {
+    return 1;
+  }
+
+  /* 198.51.100.0/24 */
+  uip_ipaddr(&ipaddr, 198, 51, 100, 0);
```

Comment:
I think that some code doesn't need to be added here.  I think that some comments will be enough.  They should say that those checks depend on the `uip_ipaddr(&mask, 255, 255, 255, 0);` from the previous check.